supabase/functions/cron_email/index.ts

```diff
@@ -0,0 +1,156 @@
+import { serve } from 'https://deno.land/std@0.199.0/http/server.ts'
+import { getEnv, methodJson, sendRes } from '../_utils/utils.ts'
+import type { BaseHeaders } from '../_utils/types.ts'
+import { trackEvent } from '../_utils/plunk.ts'
+import { supabaseAdmin } from '../_utils/supabase.ts'
+
+// This is required propoably due to this https://github.com/supabase/postgrest-js/issues/408
+interface AppWithUser {
+  app_id: string
+  name: string | null
+  user_id: {
+    email: string
+  }
+}
+
+const thresholds = {
+  // Number of updates in plain number
+  updates: [
+    100,
+    1000,
+    10000,
+  ],
+  // Percentage in decimal form (0.9 ==== 90%)
+  failRate: [
+    0.80,
+    0.90,
+    0.95,
+  ],
+  // Number of app opens in plain number
+  appOpen: [
+    500,
+    1500,
+    5000,
+  ],
+}
+
+const funComparisons = {
+  updates: [
+    'a cupcake to every student in a small school!',
+    'a pizza to every resident of a small town!',
+    'a burger to everyone in a big city!',
+  ],
+  failRate: [
+    'Even cats don\'t land on their feet that often!',
+    'That\'s a success rate higher than the average pass rate of a tough university exam!',
+    'That\'s a success rate that even the best basketball players would envy!',
+  ],
+  appOpen: [
+    'Your app was opened more times than a popular local bakery\'s door!',
+    'Your app was more popular than the latest episode of a hit TV show!',
+    'Your app was opened more times than a blockbuster movie on its opening weekend!',
+  ],
+}
+
+// Check what treshold does the stat qualify for and return the fun comparison
+function getFunComparison(comparison: 'updates' | 'failRate' | 'appOpen', stat: number): string {
+  const thresholdsForComparisons = thresholds[comparison]
+  const index = thresholdsForComparisons.map((threshold, index) => {
+    if (threshold >= stat)
+      return index
+    else if ((index === 2 && stat >= threshold))
+      return 2 // Last index
+
+    return undefined
+  }).find(i => i !== undefined)
```

Comment:
Instead of map + find, we could use [`findIndex`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) here

Reply:
Changed [here](https://github.com/Cap-go/capgo/pull/359/commits/6092f16622d27cf99f4ef74fee974bbd583e5c94)